tests/test_events.py

```diff
@@ -330,17 +333,22 @@ def test_event_is_upcoming(event, mocker):
         assert not test_event.is_upcoming
 
 
-def test_delete_duplicate_event(event, requests_mock):
+def test_delete_duplicate_event(event, client):
     # create 1 event with a fake api_source and use requests_mock to come up with 2 different responses (an ok and a 404)
     e = event.build()
-    event_url = 'http://webapi.legistar.com/v1/metro/events/0000'
+    slug = e.slug.int
+    # event_template = reverse('events', args=[slug]) # this is throwing a NoReverseMatch error despite my double checking the url pattern name
+    event_url = 'http://webapi.legistar.com/v1/metro/events/{slug}'
```

Comment:
This was tricky. This was failing with the exception:

```
(Pdb) reverse('events', args=[e.slug])
*** django.urls.exceptions.NoReverseMatch: Reverse for 'events' with arguments '(UUID('2b2e72d4-64a2-40a9-945d-0ee7a63e07a8'),)' not found. 1 pattern(s) tried: ['events/$']
```

The thing that stood out to me is that it was trying `events/$` (no args) instead of `event/<slug>/` as defined in `lametro/urls.py`. 

Django resolves URLs by [looking through the URLs declared in each installed app and stopping at the first match](https://docs.djangoproject.com/en/2.2/topics/http/urls/#how-django-processes-a-request). Sure enough, `councilmatic_core` [declares a URL with the name `events`](https://github.com/datamade/django-councilmatic/blob/66e95bae035a635a65d39d74df3be35315c14d83/councilmatic_core/urls.py#L40) that does not accept a `slug` argument, and `lametro` [overrides it](https://github.com/datamade/la-metro-councilmatic/blob/23e1dccf8c60f7cf2c3f5624c05b28b233b7e25c/councilmatic/urls.py#L35). Django's resolver is encountering the `councilmatic_core` URL first, so it stops there and fails.

Luckily, Django allows for [namespacing URL lookups](https://docs.djangoproject.com/en/2.2/topics/http/urls/#reversing-namespaced-urls). In other words, we can tell it to look specifically for the URL named `events` in the `lametro` app. 

Adding the namespace to the URL resolves this issue.

```suggestion
    event_url = reverse('lametro:events', args=[e.slug]) 
```

Reply:
thanks! I'll implement this when I pick this back up next week.